clients/TypeScript/packages/client/src/StateQuery/StateQueryClient.ts

```diff
@@ -47,119 +48,118 @@ export interface StateQueryClient {
-          if (response.reflection.requestId !== requestId) { return }
-          if ('AcquireSuccess' in response.result) {
-            return resolve({
-              context,
-              currentEpoch: () => {
-                ensureSocketIsOpen(socket)
-                return currentEpoch(context)
-              },
-              currentProtocolParameters: () => {
-                ensureSocketIsOpen(socket)
-                return currentProtocolParameters(context)
-              },
-              delegationsAndRewards: (stakeKeyHashes) => {
-                ensureSocketIsOpen(socket)
-                return delegationsAndRewards(stakeKeyHashes, context)
-              },
-              eraStart: () => {
-                ensureSocketIsOpen(socket)
-                return eraStart(context)
-              },
-              genesisConfig: () => {
-                ensureSocketIsOpen(socket)
-                return genesisConfig(context)
-              },
-              ledgerTip: () => {
-                ensureSocketIsOpen(socket)
-                return ledgerTip(context)
-              },
-              nonMyopicMemberRewards: (input) => {
-                ensureSocketIsOpen(socket)
-                return nonMyopicMemberRewards(input, context)
-              },
-              point,
-              proposedProtocolParameters: () => {
-                ensureSocketIsOpen(socket)
-                return proposedProtocolParameters(context)
-              },
-              release: () => {
-                ensureSocketIsOpen(socket)
-                return new Promise((resolve, reject) => {
-                  const releaseRequestId = nanoid(5)
-                  socket.once('message', (message: string) => {
-                    socket.once('close', () => {
-                      const response: Ogmios['ReleaseResponse'] = JSON.parse(message)
-                      if (response.reflection.requestId !== releaseRequestId) { return }
-                      if (response.result === 'Released') {
-                        resolve()
-                      } else {
-                        reject(new UnknownResultError(response.result))
-                      }
-                    })
-                    socket.close()
-                  })
-                  socket.send(JSON.stringify({
-                    ...baseRequest,
-                    methodname: 'Release',
-                    mirror: { requestId: releaseRequestId }
-                  } as Ogmios['Release']))
+    const requestId = nanoid(5)
+    socket.once('message', (message: string) => {
+      const response: Ogmios['AcquireResponse'] = JSON.parse(message)
+      if (response.reflection.requestId !== requestId) { return }
+      if ('AcquireSuccess' in response.result) {
+        return resolve({
+          context,
+          currentEpoch: () => {
+            ensureSocketIsOpen(socket)
+            return currentEpoch(context)
+          },
```

Comment:
This has nothing to do with this PR per se but just caught my attention :thinking: ...
(and I am pretty sure I raised this in the past and your answer was: doesn't matter because a new client is created for every request)

The state query client return here is "static", so it would only works for a certain time, until the acquired point expired. It would not be a major issue if requests were merely fired immediately, but by returning an intermediate object which does not re-acquire between requests we may end up in bad situations? What mechanisms are given to the client to cope with that? Is it simply that clients are encouraged to create a new client for each request :thinking: ?

Reply:
#87